staging/src/k8s.io/code-generator/cmd/validation-gen/output_tests/update_validations/primitives/zz_generated.validations.go

```diff
@@ -0,0 +1,100 @@
+//go:build !ignore_autogenerated
+// +build !ignore_autogenerated
+
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by validation-gen. DO NOT EDIT.
+
+package primitives
+
+import (
+	fmt "fmt"
+
+	validate "k8s.io/apimachinery/pkg/api/validate"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	field "k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func init() { localSchemeBuilder.Register(RegisterValidations) }
+
+// RegisterValidations adds validation functions to the given scheme.
+// Public to allow building arbitrary schemes.
+func RegisterValidations(scheme *runtime.Scheme) error {
+	scheme.AddValidationFunc((*T1)(nil), func(obj, oldObj interface{}, subresources ...string) field.ErrorList {
+		if len(subresources) == 0 {
+			return Validate_T1(obj.(*T1), oldObj.(*T1), oldObj != nil, nil)
+		}
+		return field.ErrorList{field.InternalError(nil, fmt.Errorf("no validation found for %T, subresources: %v", obj, subresources))}
+	})
+	return nil
+}
+
+func Validate_T1(obj *T1, oldObj *T1, isUpdate bool, fldPath *field.Path) (errs field.ErrorList) {
+	// field T1.S
+	var oldS *string = nil
```

Comment:
Another option to consider:  Move this inside the ephemeral func:

```
// field T1.S
errs = append(errs,
		func(obj string, old *T1, fldPath *field.Path) (errs field.ErrorList) {
			var oldVal *string = nil
			if old != nil {
				oldVal = &old.S
			}
			if isUpdate {
				errs = append(errs,
				validate.FixedResultUpdate(fldPath, &obj, oldVal, true, "T1.S, UpdateOnly")...)
			}
			return
	}(obj.S, oldObj, fldPath.Child("s"))...)
```

I am not sure it's better, but it's more encapsulated?

Reply:
We could also use bare code blocks to define variable scopes and avoid collisions, e.g.

```
{
  // code that handles field A
  var oldA ...
}
{
  // code that handles field B
  var oldB...
}
```

Reply:
IIRC golangci-lint or `go vet` complains that this is not idiomatic Go.  Super annoying because it is 100% idiomatic C :)

Let's leave it for now - it nags at me, but if I can't say why, it's just a distraction.

Reply:
FWIW, I also prefer the functional approach.  Maybe we should find a place to put some utility functions if we keep running into stuff like this.